File: maths/Codes/Area of Regular Polygon.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main() {
    cout << "\nThis code calculates the area of a given regular polygon. \nPlease enter the no of sides and length of each side. \n \n";
    int no_of_sides;
    double side_length, area;
    while (true) {
        cout << "Sides: ";
        cin >> no_of_sides;
        if (no_of_sides > 2) {
            cout << "Length: ";
            cin >> side_length;
            if (side_length > 0) {
                area = no_of_sides * (side_length * side_length) / (4 * tan(M_PI / no_of_sides));
                cout << "Area = " << area << " square units\n\n";
            }
            else {
                cout << "Length of a side cannot be negative\n";
            }
        }
        else {
             cout << "Polygon with less than 3 sides cannot be formed\n \n";
        }
    }
}

File: maths/codes/download/Area of Regular Polygon.py
from math import tan, pi

def get_integer(msg="Number: ", wrng_msg="Invalid Input"):
    try:
        i = int(input(msg))
    except ValueError:
        print(wrng_msg)
        return get_integer(msg, wrng_msg)
    else:
        return i


def get_float(msg="Number: ", wrng_msg="Invalid Input"):
    try:
        f = float(input(msg))
    except ValueError:
        print(wrng_msg)
        return get_float(msg, wrng_msg)
    else:
        return f


print("""
This code calculates the area of a given regular polygon. Please enter the no of sides and length of each side.
""")
while True:
    no_of_sides = get_integer("Sides: ")
    if no_of_sides > 2:
        side_length = get_float("Length: ")
        if side_length > 0:
            area = no_of_sides * (side_length ** 2) / (4 * tan(pi / no_of_sides))
            print(f"Area =  {area} square units")
        else:
            print('Length of a side cannot be negative')
    else:
        print('Polygon with less than 3 sides cannot be formed')
    print('')
